cmd/root.go

```diff
@@ -0,0 +1,271 @@
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/reflection"
+
+	"github.com/zcash-hackworks/lightwalletd/common"
+	"github.com/zcash-hackworks/lightwalletd/common/logging"
+	"github.com/zcash-hackworks/lightwalletd/frontend"
+	"github.com/zcash-hackworks/lightwalletd/walletrpc"
+)
+
+var cfgFile string
+var log *logrus.Entry
+var logger = logrus.New()
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "lightwalletd",
+	Short: "Lightwalletd is a backend service to the Zcash blockchain",
+	Long: `Lightwalletd is a backend service that provides a 
+         bandwidth-efficient interface to the Zcash blockchain`,
+	Run: func(cmd *cobra.Command, args []string) {
+		opts := &common.Options{
+			BindAddr:          viper.GetString("bind-addr"),
+			TLSCertPath:       viper.GetString("tls-cert"),
+			TLSKeyPath:        viper.GetString("tls-key"),
+			LogLevel:          viper.GetUint64("log-level"),
+			LogFile:           viper.GetString("log-file"),
+			ZcashConfPath:     viper.GetString("zcash-conf-path"),
+			NoTLSVeryInsecure: viper.GetBool("no-tls-very-insecure"),
+			CacheSize:         viper.GetInt("cache-size"),
+		}
+
+		fmt.Printf("Options: %#v", opts)
+
+		filesThatShouldExist := []string{
+			opts.TLSCertPath,
+			opts.TLSKeyPath,
+			opts.LogFile,
+			opts.ZcashConfPath,
+		}
+
+		for _, filename := range filesThatShouldExist {
+			if !fileExists(opts.LogFile) {
+				os.OpenFile(opts.LogFile, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0666)
+			}
+			if opts.NoTLSVeryInsecure && (filename == opts.TLSCertPath || filename == opts.TLSKeyPath) {
+				continue
+			}
+			if !fileExists(filename) {
+				os.Stderr.WriteString(fmt.Sprintf("\n  ** File does not exist: %s\n\n", filename))
+				os.Exit(1)
+			}
+		}
+
+		// Start server and block, or exit
+		if err := startServer(opts); err != nil {
+			log.WithFields(logrus.Fields{
+				"error": err,
+			}).Fatal("couldn't create server")
+		}
+	},
+}
+
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+func startServer(opts *common.Options) error {
+	if opts.LogFile != "" {
+		// instead write parsable logs for logstash/splunk/etc
+		output, err := os.OpenFile(opts.LogFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"error": err,
+				"path":  opts.LogFile,
+			}).Fatal("couldn't open log file")
+		}
+		defer output.Close()
+		logger.SetOutput(output)
+		logger.SetFormatter(&logrus.JSONFormatter{})
+	}
+
+	logger.SetLevel(logrus.Level(opts.LogLevel))
+	// gRPC initialization
+	var server *grpc.Server
+
+	if opts.NoTLSVeryInsecure {
```

Comment:
Since the `-no-tls-very-insecure` option can now be hidden inside a configuration file, please print a warning to the screen here and to the log file.

Reply:
Is the added logging sufficient? "Warn" good log level?

Reply:
Warning output added.